test/unit/test_beam_search.py

```diff
@@ -227,6 +227,8 @@ def test_update_scores():
     pad_dist = mx.nd.full((batch_beam_size, vocab_size - 1), val=np.inf, dtype='float32')
     eos_dist = mx.nd.full((batch_beam_size, vocab_size), val=np.inf, dtype='float32')
     eos_dist[:, C.EOS_ID] = 0
+    unk_dist = mx.nd.full((batch_beam_size, vocab_size), val=0, dtype='float32')
+    unk_dist[:, C.UNK_ID] = np.inf
```

Comment:
could you add a test to check whether the output of UpdateScores is as expected when prevent_unk is enabled? I think this default test should use `unk_dist = None`.

Reply:
added a new test.